drivers/interrupt_controller/intc_nxp_irqsteer.c

```diff
@@ -270,6 +275,10 @@ struct irqsteer_dispatcher {
 	uint32_t master_index;
 	/* which interrupt line is the dispatcher tied to? */
 	uint32_t irq;
+	/* reference count for all IRQs aggregated by dispatcher */
+	uint8_t irq_refcnt[CONFIG_MAX_IRQ_PER_AGGREGATOR];
+	/* dispatcher lock */
+	struct k_spinlock lock;
```

Comment:
not with crt. approach because of the boundary checks. E.g: for _irqstr_disp_put_unlocked() you'd end up with:

```
if (!atomic_get(&refcnt)) {
    return;
}

if (atomic_dec(&refcnt) == 1) {
    // do something
}
```

which is not atomic.